packages/fxa-content-server/app/scripts/views/mixins/connect-another-device-mixin.js

```diff
@@ -74,21 +74,39 @@ export default {
   /**
    * Checks to see if a user is eligible for the pairing flow.
    *
-   * To be eligible, the user must not be in a sign-in or sign-up flow,
-   * and have access CAD from the Firefox desktop app toolbar menu or app
-   * menu.
+   * Users can access the connect another device (CAD) page through various ways and we
+   * conditionally redirect them to the `/pair` page by returning `true` here.
    *
+   * Users logging in or registering for a new account will have an action=email param present,
+   * and we want to keep these users on the CAD page. We want to redirect logged in users if they
+   * get to CAD within FF via:
+   *
+   * - FF account > connect another device: entrypoint=fxa_discoverability_native
+   *   - if not signed in: can't access
+   * - about:preferences#sync > connect another device: entrypoint=preferences
+   *   - if not signed in, "sign in to sync...": entrypoint=preferences&action=email
+   * - menu > email (FF account) > connect another device: entrypoint=fxa_app_menu
+   *   - if not signed in, menu > sign in & sync: entrypoint=fxa_app_menu&action=email
+   * - sidebars (toolbar) > synced tabs > connect another device: entrypoint=tabs-sidebar
+   *   - if not signed in, "Sign in to sync": entrypoint=tabs-sidebar&action=email
+   * - synced tabs (toolbar) > connect another device: entrypoint=synced-tabs
+   *   - if not signed in, "sign in to sync...": entrypoint=synced-tabs&action=email
+   *
+   * We also want to keep users on CAD if they hit the page directly (no parameters).
    */
   isEligibleForPairing() {
     const context = this.relier.get('context');
     const entrypoint = this.relier.get('entrypoint');
-
+    const isNotActionEmail = this.relier.get('action') !== 'email';
     if (
       this.isDefault() &&
       context === Constants.FX_DESKTOP_V3_CONTEXT &&
       (entrypoint === Constants.FIREFOX_TOOLBAR_ENTRYPOINT ||
-        (entrypoint === Constants.FIREFOX_MENU_ENTRYPOINT &&
-          this.relier.get('action') !== 'email'))
+        (isNotActionEmail &&
```

Comment:
I am tempted to forego the last 4 entrypoint checks and redirect if `action=email` isn't present, _but_ we've already had a lot of small bugs come up around this stuff I don't want to introduce another one for a signin/login flow that I might not be aware of that doesn't have the param present. 🤷‍♀️

FWIW this is basically what we had prior to [this PR](https://github.com/mozilla/fxa/pull/14082/files) but with that additional check.

Reply:
It seems to me that it _should_ behave consistently, and therefore removing the last 4 checks is a good call... But then again, I've been duped by this function before, so clearly my assumptions are not the best.

By the way, there is a set of common entry points declared [here](https://github.com/mozilla/fxa/blob/fe4e4682d5011560cb810c7883fd64ad8f0da4ce/packages/fxa-content-server/app/scripts/lib/constants.js#L169). As you can see there are 7 'common' entry points, but we only reference 5 of them in this function. It stands to reason that  FIREFOX_FX_VIEW_ENTRYPOINT should behave like the other four below. FIREFOX_IOS_OAUTH_ENTRYPOINT I am less sure about…